mgmtsystem/hawkular.py

```diff
@@ -766,6 +788,372 @@ def delete_resource(self, feed_id, resource_id):
+                    child_j['children']['resource'],
+                    include_data,
+                    resource_type_id,
+                    types_index,
+                    parent_resource_id))
+        return entities
+
+    def get_config_data(self, feed_id, resource_id):
+        """Returns the data/configuration information about resource by provided
+
+        Args:
+            feed_id: Feed id of the resource
+            resource_id: Resource id
+         """
+        if not feed_id or not resource_id:
+            raise KeyError("'feed_id' and 'resource_id' are mandatory field!")
+        result = self._post_raw('strings/raw/query',
+            data={"fromEarliest": "true", "order": "DESC",
+                "tags": "feed:{},type:r,id:{}".format(feed_id, self._get_resource_id(resource_id))})
+        if result.status_code == 200:
+            entity_j = json.loads(result.content)
+            if entity_j:
+                try:
+                    inventory_structure = self._get_data_value(
+                        entity_j[0]['data'])['inventoryStructure']
+                    data_value = inventory_structure['children']['dataEntity'][0]['data']['value']
+                    return ResourceData(inventory_structure['data']['name'],
+                                        CanonicalPath(
+                                            inventory_structure['data']['resourceTypePath']),
+                                        data_value)
+                except:
+                    raise KeyError('Resource data not found for resource {} in feed '.format(
+                        resource_id, feed_id))
+        return None
+
+    def _filter_types_index(self, types_index, resource_type_id):
+        """Return values of index type which is belongs to provided resource id
+
+        Args:
+            types_index: all index types of inventory
+            resource_id: Resource id
+        """
+        entities = []
+        if resource_type_id in types_index:
+            entities.extend(types_index[resource_type_id])
+        return entities
+
+    def _get_canonical_path(self, types_index, data_id, resource_type_path, parent_resource_id):
+        """Generated the canonical path by given parameters.
+
+        Args:
+            types_index: index types for particular resource
+            data_id: ['data']['id'] field of child resource
+            resource_type_path: the path of resource type
+            parent_resource_id: id of parent Resource
+        """
+        resource_id = ''
+        data_id = self._encode_resource_id(data_id)
+        for index in types_index:
+            if index.endswith(data_id):
+                """ take full resource_id if it is listed in index types,
+                    otherwise resource itself is the parent resource
+                """
+                resource_id = index
+        # format resource id to be used in canonical path
+        if resource_id:
```

Comment:
Since ```resource_id``` is at going to be ```''``` at worst and the value of ```index``` at best, we may be able to remove this conditional completely